pennylane/ops/qubit.py

```diff
@@ -565,6 +567,96 @@ def decomposition(phi, theta, omega, wires):
         return decomp_ops
 
 
+class PauliRot(Operation):
+    r"""PauliRot(theta, pauli_word, wires)
+    Arbitrary pauli word rotation.
+
+    .. math::
+
+        RP(\theta, P) = \exp(-i \frac{\theta}{2} P)
+
+    **Details:**
+
+    * Number of wires: Any
+    * Number of parameters: 2 (1 differentiable parameter)
+    * Gradient recipe: :math:`\frac{d}{d\theta}f(RP(\theta)) = \frac{1}{2}\left[f(RP(\theta +\pi/2)) - f(RP(\theta-\pi/2))\right]`
+      where :math:`f` is an expectation value depending on :math:`RP(\theta)`.
+
+    .. note::
+
+        If the ``PauliRot`` gate is not supported on the targeted device, PennyLane
+        will attempt to decompose the gate using :class:`~.RX`, :class:`~.Hadamard`, :class:`~.RZ`
+        and :class:`~.CNOT` gates.
+
+    Args:
+        theta (float): rotation angle :math:`\theta`
+        pauli_word (string): the Pauli word defining the rotation
+        wires (Sequence[int] or int): the wire the operation acts on
+    """
+    num_params = 2
+    num_wires = Any
+    do_check_domain = False
+    par_domain = "R"
+    grad_method = "A"
+
+    _PAULI_CONJUGATION_MATRICES = {
+        "X": Hadamard._matrix(),
+        "Y": RX._matrix(np.pi / 2),
+        "Z": np.eye(2),
+    }
+
+    @staticmethod
+    def _matrix(*params):
+        theta = params[0]
+        pauli_word = params[1]
+        # We keep imaginary wires to be able to reconstruct the position of the identities
```

Comment:
I'm slightly confused by this comment, is it not the case that these wires are left out from `active_wires`?

Reply:
This refers to the fact that in `_matrix` I have no concept of wires. But because I construct `IXIYIZ` by first making `XYZ` and then expanding it to `IXIYIZ`, I have to store the information on which wires the `XYZ` are. And therefore I make up `wires` that do not a priori exist when `_matrix` is called.

Reply:
Maybe I should rename it